components/build-templates/base/e2e/serviceaccount.yaml

```diff
@@ -2,9 +2,9 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: appstudio-pipeline
-  namespace: build-templates-e2e
```

Comment:
I think this is needed, otherwise it will be created in the build-templates namspace

Also there was a reason why we did not keep the service account in base, but duplicated it for production and stage. The reason is that it the SA file will then be deployed to dev env as well. But it will fail to find the secrets, causing problems.

Reply:
> I think this is needed, otherwise it will be created in the build-templates namspace

namespace is specified in the `kustomization.yaml`

Reply:
I see that, but in the second part of the comment I mentioned that the SA should not be put into base, as it will try to deploy it to the dev env as well. In stage/prod kustomization files, the namespace is not mentioned. It loads the ../base kustomization file first, not the ../base/e2e so the namespace is ambiguous and does not work (or at least it didn't previously - output of kustomize build also does not show the correct namespace in that case)

Reply:
https://github.com/redhat-appstudio/infra-deployments/commit/b7ac9be6d926e1bc35ccd8a912411acc3016c99f

Reply:
Updated.

Reply:
After your change the commit message is not up to date. This part 
```
All above are done for both production and staging. As a result, the
production and staging serviceaccount YAML files are same, then move it
to base/e2e/ to remove the dupliate.
```
Is no longer valid.

And a little nitpick, as in this case it does not matter because the commits in the PR will be squashed before they are merged. You made a change in the 1 commit but reversed it in the second commit. For future, simple amend or rebase would be better for reviewers.

Reply:
> You made a change in the 1 commit but reversed it in the second commit. For future, simple amend or rebase would be better for reviewers.

:100: